### PR TITLE
Add JSON schemas

### PR DESCRIPTION
**Pull request by twalmsley**
This review is so you know what has been produced, although it would be handy if someone could learn JSON schema and check my work as I only learned enough to get me by.

Once we're happy with this I can generate the Java class bindings for both modules for @quentin-num to use.

- Added JSON schemas for modules 1 and 3 - they will be publicly available at `https://modules.numprotocol.com/1/schema.json` etc.
- Added test MODL files for both modules - these were sampled from our `test` environment database, so might not cover all possible variations.
- Added **manual** test scripts for both modules - converts the MODL to JSON using the latest RCFs and validates against the schemas.

Running the tests: Ideally someone would use this branch to run the test scripts as part of a review.
- Install `nodejs`, `npm`, and `ajv`
- Run `schema-tests-module-1.sh`
- Run `schema-tests-module-3.sh`
- There _shouldn't_ be any errors logged.

Still to do:

- Schemas and tests for other modules.
- Integrate the test scripts into the CI process (the scripts currently depend on `nodejs`, `npm`, and `ajv` which would need to be installed during the build, or use a different schema validator).

### CODE REVIEW

**Review by quentin-num — APPROVED** [review]
looks good, I like that you've added tests. Would be good to consider running them in jenkins as well down the line.

also would be good to add a validation of the schema files as part of the jenkins test at some point later (to pick up syntax errors, typos and such)

**Comment by twalmsley** [comment]
> looks good, I like that you've added tests. Would be good to consider running them in jenkins as well down the line.

Already mentioned in the PR description.

> 
> also would be good to add a validation of the schema files as part of the jenkins test at some point later (to pick up syntax errors, typos and such)

If the schema is invalid then the tests will fail when they're integrated into the CI pipeline.

File: data/1/schema.json [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+			],
+			"additionalProperties": false
+		},
+		"address": {
+			"$id": "#root/address",
+			"title": "address",
+			"type": "object",
+			"default": {},
+			"properties": {
+				"value": {
```
Comment (elliottinvent) [inline data/1/schema.json:108]: Address can't have a `value`

**Review by elliottinvent — CHANGES_REQUESTED** [review]

File: data/1/schema.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (twalmsley) [reply data/1/schema.json]: Good spot - removed it.

**Review by elliottinvent — APPROVED** [review]
JSON schema looks good from my basic understanding of how it works, I'll learn more about it when I can. I haven't run the tests, but we need to merge this so we can create the bindings and Q can start his work.

**Comment by twalmsley** [comment]
The Java bindings were being generated incorrectly so I've had to update the contacts schema to list methods explicitly - facebook, twitter, etc. 
It means that new methods added to the RCF will also need to be added to the schema.

**Pull request merged**